src/components/Reservation.js

```diff
@@ -1,9 +1,187 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { Navigate } from 'react-router-dom';
+import moment from 'moment';
+import { Form, Button } from 'react-bootstrap';
+import { useSelector, useDispatch } from 'react-redux';
+import styles from '../styles/reservation.module.css';
+import postReservationThunk from '../helpers/postReservation';
+
+export default function Reservation(props) {
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [submitted, setSubmitted] = useState(false);
+  let { destination } = props;
+  const destinationPassed = destination !== 'default';
+  destination = destination === 'default' ? 'Bora Bora' : destination;
+  const [chosenDestination, setChosenDestination] = useState(destination);
+  const dispatch = useDispatch();
+
+  const username = useSelector((state) => state.user.username);
+  const token = useSelector((state) => state.user.token);
+
```

Comment:
Hey Yen your username is undefined. I believe that is the reason why I can't see the username auto filled. 
When I modify this peace of code
![image](https://user-images.githubusercontent.com/63932912/142607512-f62766e5-9b64-4119-941c-ed0c7da40180.png)

I am getting this in console 
![image](https://user-images.githubusercontent.com/63932912/142607631-ca8e47de-4d71-4881-aa7d-81050f46c7fa.png)

Reply:
Hi @ShinobiWarior thanks for pointing it out. I tried to fix it. But after that, right now I have the problem of the authentication token expired on the server already when the post request gets sent.